### PR TITLE
Various small cleanups in shared code

### PR DESCRIPTION
**Pull request by keszybz**
Small cleanups and follow-up for #3131.

### CODE REVIEW

File: src/basic/dirent-util.c [hunk 1]
```diff
@@ -66,12 +64,10 @@ bool dirent_is_file(const struct dirent *de) {
...
         assert(de);
 
-        if (de->d_type != DT_REG &&
-            de->d_type != DT_LNK &&
-            de->d_type != DT_UNKNOWN)
+        if (!IN_SET(de->d_type, DT_REG, DT_LNK, DT_UNKNOWN))
                 return false;
 
-        if (hidden_file_allow_backup(de->d_name))
+        if (de->d_name[0] == '.')
```
Comment (poettering) [inline src/basic/dirent-util.c:70]: Well, strictly speaking the function won't filter out "aquota.user" anymore, if people check for ".user" as suffix... Probably doesn't matter, but I just wanted to mention it...

Comment (keszybz) [inline src/basic/dirent-util.c:70]: True. But yeah, shouldn't matter unless you're looking for *.user in a root of some mount point.

**Comment by keszybz** [comment]
Should be less broken now.

**Comment by poettering** [comment]
So, a228a22fda4faa9ecb7c5a5e499980c8ae5d2a08 split up the two hidden calls, because we wanted a nice way to enumerate journal files suffixed with "~"... I think this PR will break that again?

**Comment by keszybz** [comment]
I see. I was confused by "allow backup" — for me .rpmnew and .rpmsave and so on are also backup files. So I thought the function name makes no sense. It does, kind of, but the new name is still better.

> split up the two hidden calls, because we wanted a nice way to enumerate journal files split up the two hidden calls, because we wanted a nice way to enumerate journal files 

The journal code calls `dirent_is_file_with_suffix`, but ddb3706 removes the call to hidden_file. So there's no need to have the split anymore. After all, if we filter by ending (*.journal[~]), then we don't need to blacklist other extensions.

**Pull request merged**